src/core/idebugsettings.cpp

```diff
@@ -0,0 +1,7 @@
+/* SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright © 2024 The TokTok team.
+ */
+
+#include "idebugsettings.h"
+
+IDebugSettings::~IDebugSettings() = default;
```

Comment:
Why the out-of-line d-tor?

Reply:
For vtables.

Reply:
Ah, right. https://clang.llvm.org/docs/DiagnosticsReference.html#wweak-vtables